pkg/apis/core/types.go

```diff
@@ -413,6 +413,11 @@ type PersistentVolumeClaimSpec struct {
 	// This is an alpha feature and may change in the future.
 	// +optional
 	VolumeMode *PersistentVolumeMode
+	// If specified, volume will be prepopulated with data from the specified data source.
+	// This depends on the provisioner for this volume being able to use the specified source.
+	// If the provisioner does not support it, it will fail to provision the volume.
+	// +optional
+	DataSource *TypedLocalObjectReference
```

Comment:
Need more details here, see Tim's comment in the proposal https://github.com/kubernetes/community/pull/2335#discussion_r213037163

maybe something like 
// This is alpha feature and may change in the future
// It requires the VolumeSnapshotDataSource alpha feature gate to be enabled and DataSource  must be from VolumeSnapshot

Reply:
Done